File: advent_of_code/day1.js
const fs = require('fs')
const path = require('path')

function sumFrequencyChanges(file) {
  let frequency_changes = fs.readFileSync(file).toString().split('\n')
  let sum = 0
  for (let i of frequency_changes) {
    sum += parseInt(i)
  }
  return sum
}

console.log(sumFrequencyChanges(path.join(__dirname, '/day1.txt')))

File: advent_of_code/day1.py
import os

def sum_frequency_changes(file):
    sum = 0
    with open (file, 'r') as f:
        for line in f:
            sum += int(line)
    return sum

dir_path = os.path.dirname(os.path.abspath(__file__))
print(sum_frequency_changes(os.path.join(dir_path, 'day1.txt')))
